src/Serialization.purs

```diff
@@ -0,0 +1,187 @@
+module Serialization
+  ( convertTransaction
+  , addressPubKeyHash
+  , convertBigInt
+  , convertTxOutput
+  , newAddressFromBech32
+  , newAssetName
+  , newTransactionHash
+  , toBytes
+  , newBaseAddressFromAddress
+  , module X
+  )
+where
+
+import Data.BigInt as BigInt
+import Data.Maybe (Maybe(..))
+import Data.Newtype (unwrap)
+import Data.Traversable (traverse_, for_)
+import Data.FoldableWithIndex (forWithIndex_)
+import Data.UInt (UInt)
+import Effect (Effect)
+import Prelude
+import Serialization.Types (Address, AssetName, Assets, AuxiliaryData, BaseAddress, BigNum, DataHash, Ed25519KeyHash, Ed25519Signature, MultiAsset, PlutusScript, PlutusScripts, PublicKey, ScriptHash, StakeCredential, Transaction, TransactionBody, TransactionHash, TransactionInput, TransactionInputs, TransactionOutput, TransactionOutputs, TransactionWitnessSet, Value, Vkey, Vkeywitness, Vkeywitnesses)
+import Serialization.Types (Address, AssetName, Assets, AuxiliaryData, BaseAddress, BigNum, DataHash, Ed25519KeyHash, Ed25519Signature, MultiAsset, PlutusScript, PlutusScripts, PublicKey, ScriptHash, StakeCredential, Transaction, TransactionBody, TransactionHash, TransactionInput, TransactionInputs, TransactionOutput, TransactionOutputs, TransactionWitnessSet, Value, Vkey, Vkeywitness, Vkeywitnesses) as X
+import Types.ByteArray (ByteArray)
+import Types.Transaction as T
+import Untagged.Union (type (|+|))
+
+-- * cardano-serialization-lib types
+
+foreign import newBigNum :: String -> Effect BigNum
```

Comment:
Do we want to follow the convention from other FFI code of using a `_` prefix for `foreign import`s? (Genuine question, I'm not sure how readable or desirable that would be in this case)

Reply:
i think in general this is good practice.

Reply:
it's a big alarm bell that 'the type is possibly wrong'

Reply:
@Benjmhart but defining a new function without underscore that just reuses the FFI one does not add more guarantees on that front.

I thought that the convention is to use `_` for functions that accept extra arguments or are "implementations" in some other sense.